ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
This doesn't look right, it breaks the implicit result dropping subsumption that is allowed for all other expressions. For example, you could no longer use a `br_if` with argument in a block, as your test cases already demonstrate:
```
(block (br_if $l (arg) (cond)) (other))
```
becomes a type error. That violates a basic design principle of the type system, and doesn't make sense from a practical perspective.

However, the previous version of this PR changed the typing of `br` and `return`. As @JSStats pointed out, it effectively reverted PR #180, which addressed issue #179. We could work around that with more special casing, but that would be fairly ugly, and I'm not convinced it's worth it.

I think the least inconsistent way to implement this change is to reach agreement that we undo #180. @jcbeyler, @lukewagner, would you be okay with that? It's not ideal either, but at this point, I don't have a particularly strong opinion about that one. It's still much better than introducing special case rules for optional arguments, which I'd very much like to avoid.